core/src/components/segment/segment.tsx

```diff
@@ -76,18 +75,15 @@ export class Segment implements ComponentInterface {
   /**
    * the value of the segment.
    */
-  @Prop({ mutable: true }) value?: string | null;
```

Comment:
Why did we remove `null`?

Reply:
After reviewing the internals of the `ion-segment` component, I could not find a reason why the `value` prop accepts `null` - as nothing from segment appears to assign a value of `null`. The change was made as part of this commit: https://github.com/ionic-team/ionic-framework/commit/ba2230510e95b31d8a6a6ab103ca09888f8237b5#diff-7e01db7218b2b21c70a8e558819e7cdb4f6b6220459509b3bd798371a6555277

There were also a few inconsistencies with the types:
-  `ionChange` emits a type of `string | undefined`, but the value type is `string | undefined | null`. 
-  The types for `newValue` and `previousValue` for the watch callback: https://github.com/ionic-team/ionic-framework/blob/main/core/src/components/segment/segment.tsx#L82

This resulted in type errors when trying to emit the `value` against `ionChange`:

e.g.:
```
const { value } = this;
this.ionChange.emit({ value });
```

> Type 'string | null | undefined' is not assignable to type 'string | undefined'.
  Type 'null' is not assignable to type 'string | undefined'.

Unless type narrowing or type casting (like the previous implementation did with the `@Watch` callback function. 

I'm open to feedback on this though - if we know a reason that `null` should be an accepted value type.

Reply:
After doing more research into this, I think the original intent of `null` was to offer developers a way to clear the checked item in the segment. 

I will update the breaking change docs to reflect this, but assigning a value of `''` would be preferred in this use case.